src/functions-reference/complex_matrix_operations.Rmd

```diff
@@ -453,5 +692,406 @@ The sum of the values in x, or 0 if x is empty
 
 `complex` **`sum`**`(complex_matrix x)`<br>\newline
 The sum of the values in x, or 0 if x is empty
-`r since("2.0")`
+`r since("2.30")`
+
+<!-- complex; prod; (complex_vector x); -->
+\index{{\tt \bfseries prod }!{\tt (complex\_vector x): complex}|hyperpage}
+
+`complex` **`prod`**`(complex_vector x)`<br>\newline
+The product of the values in x, or 1 if x is empty
+`r since("2.30")`
+
+<!-- complex; prod; (complex_row_vector x); -->
+\index{{\tt \bfseries prod }!{\tt (complex\_row\_vector x): complex}|hyperpage}
+
+`complex` **`prod`**`(complex_row_vector x)`<br>\newline
+The product of the values in x, or 1 if x is empty
+`r since("2.30")`
+
+<!-- complex; prod; (complex_matrix x); -->
+\index{{\tt \bfseries prod }!{\tt (complex\_matrix x): complex}|hyperpage}
+
+`complex` **`prod`**`(complex_matrix x)`<br>\newline
+The product of the values in x, or 1 if x is empty
+`r since("2.30")`
+
+
+## Vectorized accessor functions
+
+Much like with complex scalars, two functions are
+defined to get the real and imaginary components of complex-valued
+objects. These functions return the same shape (e.g matrix, vector, row vector,
+or array) object as their input, but
+demoted to a `real` type. For example, `get_real(complex_matrix M)` yields a
+`matrix` containing the real component of each value in `M`.
+
+<!-- T; get_real; (T x); -->
+\index{{\tt \bfseries get\_real }!{\tt (T x): T\_demoted}|hyperpage}
+
+`T_demoted` **`get_real`**`(T x)`<br>\newline
+Given an object of complex type `T`, return the same shape object but of type
```

Comment:
I think you should call out a special section in which to explain the `T_demoted` terminology.   It can just be a subsection of this section (use `###` or `####`), but it should be linkable from the definitions using `T_demoted`.  And I'd like some more examples (`complex_matrix` to `matrix` and `array[] complex` to `array[] real` and `complex` to `real`).

I think I'd like `real<T>` better than `T_demoted` because it indicates the target basis type, whereas `_demoted` is very vague.  What do you think?

Reply:
I think that usage of `<>` would be confusing. I will add some explanation of `T_demoted`